pdf/pdfrenderthread.cpp

```diff
@@ -279,19 +279,21 @@ void PDFRenderThread::cancelRenderJob(int index)
 void PDFRenderThread::prioritizeJob(int index, int size)
 {
     QMutexLocker locker(&d->thread->mutex);
-    for (QList<PDFJob *>::iterator it = d->thread->jobQueue->begin(); it != d->thread->jobQueue->end(); ++it) {
+    RenderPageJob *last = nullptr;
+    for (QList<PDFJob *>::iterator it = d->thread->jobQueue->begin(); it != d->thread->jobQueue->end();) {
         PDFJob *j = *it;
         if (j->type() == PDFJob::RenderPageJob) {
             RenderPageJob *rj = static_cast<RenderPageJob *>(j);
             if (rj->m_index == index && rj->renderWidth() == size) {
-                if (it != d->thread->jobQueue->begin()) { // If it is already at the front, just abort..
-                    d->thread->jobQueue->erase(it);
-                    d->thread->jobQueue->push_front(j);
-                }
-                return;
+                it = d->thread->jobQueue->erase(it);
+                last = rj;
```

Comment:
You're right and this prioritize stuff is still not optimal. Currently for instance, when scrolling fast to the page, let say 5, the rendering queue is filled up with render(3), render(4), render(5), re,der(6), render(7) ans the 5th page is not in the top of the list.

I will try to remove this prioritize() function in favor of a priotirtize argument to the requestPage(). Then renderPage(i) do the following:
* if page i is already in the queue, waiting, then its subpart is updated with the new one; otherwise it is appended as a new request.
* if the priority argument is set, then the job in the queue is moved to the top.